source/common/common/logger.h

```diff
@@ -358,4 +363,80 @@ template <Id id> class Loggable {
 
 // TODO(danielhochman): macros(s)/function(s) for logging structures that support iteration.
 
+/**
+ * ---------------------------------------------------------------
+ * Fancy logging macros
+ */
+
+using FancyMap = absl::flat_hash_map<std::string, std::shared_ptr<spdlog::logger>>;
+using FancyMapPtr = std::shared_ptr<FancyMap>;
+using SpdLoggerPtr = std::shared_ptr<spdlog::logger>;
+
+extern spdlog::level::level_enum kFancyDefaultLevel;
+
+extern absl::Mutex fancy_log_lock__;
+
+extern FancyMapPtr getFancyLogMap();
+
+extern void setFancyLogger(std::string key, spdlog::level::level_enum log_level);
+
+extern void initFancyLogger(std::string key, std::atomic<spdlog::logger*>& logger);
+
+extern spdlog::sink_ptr getSink();
+
+#define FANCY_KEY std::string(__FILE__)
```

Comment:
How do you plan to use these commented out macros? Why are they commented out?

Reply:
I commented out them because I want to remain a simplest solution for finer control in code (directly uncomment one of those) without going too far. The ideal solution should be compile options of file, function or line level control.